src/ui/CategoryFacet/CategoryFacet.ts

```diff
@@ -858,11 +921,14 @@ export class CategoryFacet extends Component implements IAutoLayoutAdjustableIns
   }
 
   private handleQueryStateChanged(data: IAttributesChangedEventArg) {
-    if (this.listenToQueryStateChange) {
-      let path = data.attributes[this.queryStateAttribute];
-      if (!Utils.isNullOrUndefined(path) && isArray(path) && path.length != 0) {
-        this.changeActivePath(path)
-      }
+    if (!this.listenToQueryStateChange) {
+      return;
+    }
+
+    const path = data.attributes[this.queryStateAttribute];
+    if (!Utils.isNullOrUndefined(path) && isArray(path) && path.length != 0) {
+      this.changeActivePath(path);
+      this.selectPath(path as string[]);
```

Comment:
Why do we need to cast here ?
Is it hiding a problem in the types of `data.attributes[this.queryStateAttribute];`